### ISSUE TITLE
Symbol.iterator fails when decorators and class-properties are used in Babel 7

### ISSUE THREAD

**Issue by AlexanderOMara** [babel/babel#7705]
**Bug report**

A class that implements `Symbol.iterator` when a decorator is used on a class property produces the following error when running the transpiled code:

> Error: Class "Symbol" cannot be referenced in computed property keys.

[Gist to demonstrate the issue](https://gist.github.com/AlexanderOMara/58ab0e943a441120758dbe472fcb47db)

### Input Code

```js
function property(enumerable = true, configurable = true, writable = true) {
	return (target, key, descriptor) => {
		descriptor.enumerable = enumerable;
		descriptor.configurable = configurable;
		descriptor.writable = writable;
	};
}

class Example extends Object {

	@property()
	test;

	constructor() {
		super();
	}

	[Symbol.iterator]() {
		return {
			next: function() {
				return {done: true};
			}
		};
	}
}
```

### Babel/Babylon Configuration (.babelrc, package.json, cli command)

```js
{
	"presets": [],
	"plugins": [
		["@babel/plugin-proposal-decorators"],
		["@babel/plugin-proposal-class-properties", {
			"loose" : true
		}]
	]
}
```

[See transpiled code here](https://gist.github.com/AlexanderOMara/58ab0e943a441120758dbe472fcb47db#file-index-js), in particular this part:

```js
_iterator = (_classNameTDZError("Symbol"), Symbol).iterator
```

### Expected Behavior

Expected `Symbol.iterator` to function normally, as it does when there are no decorated class-properties.

### Current Behavior

Transpiled code unconditionally throws an error when these conditions are met.

### Context

Just trying to use `Symbol.iterator`, class `@babel/plugin-proposal-decorators`, and `@babel/plugin-proposal-class-properties` at the same time.

It's possible to work around by aliasing `Symbol.iterator` with a local variable, but that's very unpleasant, especially for TypeScript code (other TypeScript parsing code cannot infer the type properly then).

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  7.0.0-beta.41
| node             | v9.11.1
| npm              | 5.6.0
| Operating System | macOS 10.13.4

**Comment by jsg2021** [comment]
I got hit by this today.

| software         | version(s)
| ---------------- | -------
| Babel            |  7.0.0-beta.44
| node             | 9.8.0
| npm              | 5.8.9
| Operating System | macOS 10.13.4

**Comment by mmantel** [comment]
Looks like there is an open PR to fix this issue, #7475.

**Comment by jsg2021** [comment]
I still get this in beta.51. I haven't been able to use any of the betas because of this issue. :(
